build-go.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# borrowed from https://github.com/openshift/origin/blob/master/hack/lib/constants.sh os::build::ldflags()
+
+function bld_cmd() {
+	OS_ROOT=$(dirname "${BASH_SOURCE}")
+	local git=(git --work-tree "${OS_ROOT}")
+
+	if [[ -n ${OS_GIT_COMMIT-} ]] || OS_GIT_COMMIT=$("${git[@]}" rev-parse --short "HEAD^{commit}" 2>/dev/null); then
+		if [[ -z ${OS_GIT_TREE_STATE-} ]]; then
+			# Check if the tree is dirty.  default to dirty
+			if git_status=$("${git[@]}" status --porcelain 2>/dev/null) && [[ -z ${git_status} ]]; then
+				OS_GIT_TREE_STATE="clean"
+			else
+				OS_GIT_TREE_STATE="dirty"
+			fi
+		fi
+		# Use git describe to find the version based on annotated tags.
+		if [[ -n ${OS_GIT_VERSION-} ]] || OS_GIT_VERSION=$("${git[@]}" describe --long --tags --abbrev=7 --match 'v[0-9]*' "${OS_GIT_COMMIT}^{commit}" 2>/dev/null); then
+			# Try to match the "git describe" output to a regex to try to extract
+			# the "major" and "minor" versions and whether this is the exact tagged
+			# version or whether the tree is between two tagged versions.
+			if [[ "${OS_GIT_VERSION}" =~ ^v([0-9]+)\.([0-9]+)\.([0-9]+)(\.[0-9]+)*([-].*)?$ ]]; then
+				OS_GIT_MAJOR=${BASH_REMATCH[1]}
+				OS_GIT_MINOR=${BASH_REMATCH[2]}
+				OS_GIT_PATCH=${BASH_REMATCH[3]}
+				if [[ -n "${BASH_REMATCH[5]}" ]]; then
+					OS_GIT_MINOR+="+"
+				fi
+			fi
+
+			# This translates the "git describe" to an actual semver.org
+			# compatible semantic version that looks something like this:
+			#   v1.1.0-alpha.0.6+84c76d1-345
+			OS_GIT_VERSION=$(echo "${OS_GIT_VERSION}" | sed "s/-\([0-9]\{1,\}\)-g\([0-9a-f]\{7,40\}\)$/\+\2-\1/")
+			# If this is an exact tag, remove the last segment.
+			OS_GIT_VERSION=$(echo "${OS_GIT_VERSION}" | sed "s/-0$//")
+			if [[ "${OS_GIT_TREE_STATE}" == "dirty" ]]; then
+				# git describe --dirty only considers changes to existing files, but
+				# that is problematic since new untracked .go files affect the build,
+				# so use our idea of "dirty" from git status instead.
+				OS_GIT_VERSION+="-dirty"
+			fi
+		fi
+	fi
+
+	go build \
+	-ldflags "-X github.com/openshift/cluster-samples-operator/pkg/version.majorFromGit=${OS_GIT_MAJOR} \
+	-X github.com/openshift/cluster-samples-operator/pkg/version.majorFromGit=${OS_GIT_MAJOR} \
```

Comment:
you're passing major twice

Reply:
and not passing patch

Reply:
so no setting of patch, or how to gather, in the openshift origin examples

removed duplicate major